src/util/simd/test_avx_common.c

```diff
@@ -415,6 +415,73 @@ int wv_test( wv_t v, ulong v0, ulong v1, ulong v2, ulong v3 ) {
   return 1;
 }
 
+int wh_test( wh_t h, ushort const * hj ) {
+  int volatile _[1];
+  ushort       m[151] W_ATTR;
+  wh_t         l;
+
+  if( wh_extract( h,  0 )!=hj[ 0] ) return 0;
+  if( wh_extract( h,  1 )!=hj[ 1] ) return 0;
+  if( wh_extract( h,  2 )!=hj[ 2] ) return 0;
+  if( wh_extract( h,  3 )!=hj[ 3] ) return 0;
+  if( wh_extract( h,  4 )!=hj[ 4] ) return 0;
+  if( wh_extract( h,  5 )!=hj[ 5] ) return 0;
+  if( wh_extract( h,  6 )!=hj[ 6] ) return 0;
+  if( wh_extract( h,  7 )!=hj[ 7] ) return 0;
+  if( wh_extract( h,  8 )!=hj[ 8] ) return 0;
+  if( wh_extract( h,  9 )!=hj[ 9] ) return 0;
+  if( wh_extract( h, 10 )!=hj[10] ) return 0;
+  if( wh_extract( h, 11 )!=hj[11] ) return 0;
+  if( wh_extract( h, 12 )!=hj[12] ) return 0;
+  if( wh_extract( h, 13 )!=hj[13] ) return 0;
+  if( wh_extract( h, 14 )!=hj[14] ) return 0;
+  if( wh_extract( h, 15 )!=hj[15] ) return 0;
+
+  for( int j=0; j<16; j++ ) { _[0]=j; if( wh_extract_variable( h, _[0] )!=hj[j] ) return 0; }
+
+  wh_st(  m,     h ); /*   Aligned store to aligned   */
+  wh_stu( m+16,  h ); /* Unaligned store to aligned   */
+  wh_stu( m+33,  h ); /* Unaligned store to aligned+1 */
+  wh_stu( m+50,  h ); /* Unaligned store to aligned+2 */
+  wh_stu( m+67,  h ); /* Unaligend store to aligned+3 */
+  wh_stu( m+84,  h ); /* Unaligned store to aligned+4 */
+  wh_stu( m+101, h ); /* Unaligned store to aligned+5 */
+  wh_stu( m+118, h ); /* Unaligned store to aligned+6 */
+  wh_stu( m+135, h ); /* Unaligned store to aligned+7 */
+
+  l = wh_ld(  m     ); if( wc_pack( wh_eq( h, l ) )!=255 ) return 0;
```

Comment:
I don't think this is right, since `wh_eq` doesn't return a proper conditional. This will only look at 8 of the comparisons, not 16.

Reply:
Good point. I should use _mm256_movemask_epi8 instead of wc_pack here.

Reply:
Fixed